### PR TITLE
When using IP address, Blob Endpoint parser throws StringIndexOutOfBoundsException if path component is not provided

### PR DESCRIPTION
**Pull request by reta**
Minor fix for the edge case when Blob Endpoint is provided as an IP address (fe, `https://127.0.0.1:10000`), the parsing fails with `StringIndexOutOfBoundsException` when path component is empty. Sample stack trace:

```
 java.lang.StringIndexOutOfBoundsException: String index out of range: 0
	at java.lang.String.charAt(String.java:658)
	at com.azure.storage.blob.BlobUrlParts.parseIpUrl(BlobUrlParts.java:406)
	at com.azure.storage.blob.BlobUrlParts.parse(BlobUrlParts.java:372)
	at com.azure.storage.blob.BlobServiceClientBuilder.endpoint(BlobServiceClientBuilder.java:132)
	at com.azure.storage.blob.BlobServiceClientBuilder.connectionString(BlobServiceClientBuilder.java:270)
```

Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### CODE REVIEW

**Comment by rickle-msft** [comment]
This looks good to me. Can you please add an entry in the changelog to document this fix?

**Comment by reta** [comment]
> This looks good to me. Can you please add an entry in the changelog to document this fix?

@rachel-msft absolutely, thank you

File: sdk/storage/azure-storage-blob/pom.xml [hunk 1]
```diff
@@ -151,6 +151,18 @@
       <version>1.7.3</version> <!-- {x-version-update;com.azure:azure-core-http-okhttp;dependency} -->
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (reta) [inline sdk/storage/azure-storage-blob/pom.xml:154]: Noticed that JUnit tests are not run due to missing Jupiter engine

Comment (rickle-msft) [reply sdk/storage/azure-storage-blob/pom.xml:155]: I actually didn't even notice this until you pointed it out, but in Storage, we request that all tests be written in Groovy using Spock, not JUnit. If you would rather not convert them, then we can leave this open for now and I can do that when I get around to it.

Comment (reta) [reply sdk/storage/azure-storage-blob/pom.xml:155]: Oh I see, I saw a mix of JUnit / Spock across many modules. thought this is fine. I think it I could do it Spock style, sure

Comment (reta) [reply sdk/storage/azure-storage-blob/pom.xml]: Done, thanks @rickle-msft

**Comment by rickle-msft** [comment]
This looks good to me. I'm going to ask another person on our team to take a look to ensure I'm not forgetting any steps about external contributions :)

**Review by alzimmermsft — APPROVED** [review]

**Review by rickle-msft — APPROVED** [review]

**Pull request merged**